scripts/pcp/update_pcp_biotype.pl

```diff
@@ -0,0 +1,142 @@
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Updates biotype in pcp db if BOTH protein coding prediction tools agree that
+# the gene is protein coding
+
+=head1 NAME
+  update_pcp_biotype.pl
+=head1 DESCRIPTION
+Reads the output of both RNASamba and CPC2 and where they both agree that a transcript is protein coding the biotype
+is updated to pcp_protein_coding in the selected database.
+Throws if results files have non matching gene models/ number of gene models
+=head1 OPTIONS
+=head1
+=head2 Database connection options
+  -dbhost   Host name for database.
+  -dbport   What port to connect to.
+  -dbname   What name to connect to.
+  -dbuser   What username to connect as.
+  -dbpass   What password to use.
+=head2 Other options
+  --coords coordinate system name to use, default toplevel
+  --rnaSamba RNASamba output file to be used
+  --cpc2 CPC2 output file to be used
+=head1 EXAMPLES
+  ./update_pcp_biotype.pl --dbhost genebuild5 \
+    --dbname <database_name> \
+    --dbhost GBS5
+    --dport GBP5
+    --rnaSamba <RNASamba output file> \
+    --cpc2 <CPC2 output file> \
+=cut
+
+use strict;
+use warnings;
+use Getopt::Long qw(:config no_ignore_case);
+
+use Bio::EnsEMBL::Utils::Exception qw(throw);
+use Bio::EnsEMBL::DBSQL::DBAdaptor;
+
+## Some defaults
+my $coord_system = 'toplevel';
+my $dbname = '';
+my $user   = 'ensadmin';
+my $host = $ENV{GBS5};
+my $port = $ENV{GBP5};
+my $pass = 'ensembl';
+my ($cpc2_file, $rnasamba_file ) = ( '', '' );
+
+my $options = GetOptions ("user|dbuser|u=s"     => \$user,
+                          "host|dbhost|h=s"     => \$host,
+                          "port|dbport|P=i"     => \$port,
+                          "dbname|db|D=s"       => \$dbname,
+                          "dbpass|pass|p=s"     => \$pass,
+                          "cpc2=s"              => \$cpc2_file,
+                          "rnaSamba|rnas=s"     => \$rnasamba_file,
+                          "coords:s"            => \$coord_system,);
+
+my $db = new Bio::EnsEMBL::DBSQL::DBAdaptor(
+  -port    => $port,
+  -user    => $user,
+  -host    => $host,
+  -dbname  => $dbname,
+  -pass    => $pass);
+
+my %cpc_results = parse_results($cpc2_file, 8);
+my %rnasamba_results = parse_results($rnasamba_file, 2);
+my %selected_genes;
+
+if (%cpc_results ne %rnasamba_results) {
+  throw("Results files have different number of gene models\n");
+}
+else {
+  my %compare = map {$_ => 1} keys %cpc_results;
+  for my $key (keys %rnasamba_results) {
+    last unless exists $compare{$key};
+    delete $compare{$key};
+  }
+  if (%compare) {
+    throw("Results files contain different gene models\n");
+  }
```

Comment:
This all block and probably the code in "if" could probably be removed and one line added in the foreach below with something like `throw("Results files contain different gene models\n") unless (exists $rnasamba_results{$key));`

Reply:
Done.